run.py

```diff
@@ -177,10 +182,20 @@ def gh_mime_data(app):
     garb_handler_md.remove(app)
 
 
+def cache_data(app):
+    all_titles = list()
+    all_results = list()
+    w = walker.Walker(app)
+    for achievement in w.get_all_achievements():
+        all_titles.append(achievement.container.title)
+        all_results.append(achievement.result)
+
+
 def timeout_daily(app):
     log.info("daily execution handler started")
     gh_container_achievements(app)
     gh_mime_data(app)
+    cache_data(app)
```

Comment:
this example is fine, i meant he function cache_data() but should be structured a little bit different:

```
+def cache_update(app, frequency=None):
      if frequence == None:
         raise InternalError("frequenzy must be \"daily\", \"hourly\" or \"weekly\"") 

     cache_person = cache_person.Cache(frequency)
     cache_tags = cache_tags.Cache(frequency)
     cache_categories = cache_categories.Cache(frequency)
     cache_achievements = c.Cache(frequency)

     for achievement in walker.get_all_achievements():
         cache_person.update(achievement)
         cache_tags.update(achievement)
         cache_categories.update(achievement)
         cache_achievements.update(achievement)
         [...]

     cache_person.finallize()
     cache_tags.finallize()
     cache_categories.finallize()
     cache_achievements.finallize()
```

within a cache the following code can be implemented:

file: cache_tags.py
```
class Cache(object):
    def __init__(self, frequency):
          self.frequency = frequency
          self.data = dict()
          # open old cache_tags db file if required here ... 

def update(self, achievement)
          if self.frequency != "daily":
              # ok, this is the weekly or hourly timer, but we are the tags cache and
              # we have no need to update every week or hour, return gracefully here
              return

            # ok, fine, this is the daily timer, update db,
            # pseudo code is following, but similar
            self.data[achievement.container.attachments.tags] += 1

def finalize(self):
           # ok, now write new or updated data to cache_tags db file
```